consensus/src/pending_votes.rs

```diff
@@ -99,6 +100,83 @@ impl PendingVotes {
         }
     }
 
+    /// Insert a RoundTimeout and return a TimeoutCertificate if it can be formed
+    pub fn insert_round_timeout(
```

Comment:
we should re-use the code for insert_vote as well? at least majority piece

Reply:
you mean checking if there was an existing timeout vote?

Reply:
no I mean code wise it's duplicate of the piece in insert_vote?

Reply:
The return result changes between the two implementations. In the old code, we return vote voting power if we don't have enough to aggregate regardless, but in new code, we return the tc voting power. If we want to keep existing logic then we have to keep them separate.